disdrodb/data_transfer/download_data.py

```diff
@@ -0,0 +1,282 @@
+    """
+    if not os.path.exists(path):
+        raise FileNotFoundError(f"The path {path} does not exist. Please check the path.")
+
+
+def get_metadata_folders(folder_path: str) -> list:
+    """Return a list containing all metadata folders paths
+
+    Parameters
+    ----------
+    folder_path : str
+        folder path
+
+    Returns
+    -------
+    list
+        list of metadata folders
+    """
+
+    return [os.path.join(root, "metadata") for root, dirs, files in os.walk(folder_path) if "metadata" in dirs]
+
+
+def get_yaml_files(metadata_folders: str) -> list:
+    """Return a list containing all yaml files paths
+
+    Parameters
+    ----------
+    metadata_folders : str
+        folder path
+
+    Returns
+    -------
+    list
+        list of yaml files
+    """
+
+    return [
+        os.path.join(root, file)
+        for metadata_folder in metadata_folders
+        for root, dirs, files in os.walk(metadata_folder)
+        for file in files
+        if file.endswith(".yml")
+    ]
+
+
+def get_list_urls_local_paths(
+    raw_folder: str = None,
+    data_source: str = None,
+    campaign_name: str = None,
+    station_name: str = None,
+) -> list:
+    """Parse the folder according to the paramerters (data_source,
+    campaign_name and station_name) to get all url from the config files.
+
+    Returns
+    -------
+    list
+        List of tuples containing the local path and the url.
+    """
+    # test if the file path is correct and exists. If not, raise an error
+    check_path(raw_folder)
+
+    # Test that the path ends with DISDRODB/Raw. If not, raise an error
+    split_path = raw_folder.split(os.sep)
+    if split_path[-1] != "Raw" or split_path[-2] != "DISDRODB":
+        raise ValueError("The path must end with DISDRODB/Raw. Please check the path.")
+
+    # Define the path to analyse
+    path_to_analyse = raw_folder
+    if data_source:
+        path_to_analyse = os.path.join(path_to_analyse, data_source)
+    if campaign_name:
+        path_to_analyse = os.path.join(path_to_analyse, campaign_name)
+
+    # Get all config files from the metadata folders
+    metadata_folders = get_metadata_folders(path_to_analyse)
+    yaml_files = get_yaml_files(metadata_folders)
+
+    # Get the list of url and local paths from the config files
+    list_of_path_url = []
+    for yaml_file_path in yaml_files:
+        file_content = yaml.safe_load(open(yaml_file_path))
+        if "data_url" in file_content.keys():
+            data_url = file_content["data_url"]
+            config_station_name = os.path.basename(yaml_file_path).replace(".yml", "")
+            # Only if the station name is not specified or if the station name is the same as the config file name
+            if not station_name or station_name == config_station_name:
+                folder_path = os.path.dirname(yaml_file_path).replace("metadata", "data")
+                file_path = os.path.join(folder_path, config_station_name)

```

Comment:
So here we are not speaking about files paths  ... but the stations data directory path ! 
I would change the function name accordingly ! --> `get_remote_and_local_data_directories` ?
Also adapt the name of the variables

Reply:
ok